packages/dds/tree/src/test/shared-tree-core/sharedTreeCore.spec.ts

```diff
@@ -142,7 +145,9 @@ describe("SharedTreeCore", () => {
         return summaryObject.type === SummaryType.Tree;
     }
 
-    function createTree(indexes?: Index[]): SharedTreeCore<DefaultChangeset, DefaultChangeFamily> {
+    function createTree<TIndexes extends readonly Index[]>(
+        indexes: TIndexes,
```

Comment:
I see you removed the possibility of not passing an index. Was that needed for the typing changes? If not, what was the motivation?

Reply:
It required an unsafe conversion in the case when indexes are provided. It's not safe to default to `[]` anymore. The value to default to is the empty array `[]`, and while you can safely convert a `[]` to an `Index[]`, you _can't_ convert it to an arbitrary `TIndexes` even if `TIndexes` extends `Index[]`. This would happen if you called `createTree<[MockIndex]>()`, for example. `[]` does not extend `[MockIndex]`, because it's a _tuple_ with a known, required element.